### PR TITLE
Reorganize Python tests

### PR DESCRIPTION
**Pull request by tpoterba**
cc @danking @patrick-schultz @jigold 

Was anyone else part of the testing discussion? Would like input on the new structure:

image

### CODE REVIEW

File: python/test/test_hail/expr/test_expr.py [hunk 1]
```diff
@@ -1,88 +1,172 @@
-import unittest
 import math
+import unittest
 
 import hail as hl
 import hail.expr.aggregators as agg
-from hail.expr import coercer_from_dtype
 from hail.expr.types import *
-from .utils import startTestHailContext, stopTestHailContext, resource
+from ..helpers import *
```
Comment (danking) [inline python/test/test_hail/expr/test_expr.py]: what does double dot do?

Comment (tpoterba) [reply python/test/test_hail/expr/test_expr.py]: same as bash. go up a directory.

**Comment by tpoterba** [comment]
ah, I should have enumerated the changes besides motion. 

I renamed utils to helpers to avoid a name conflict. 
I moved a few things like get_dataset to helpers, which deleted a bunch of duplicated code.

I also deleted a few of the ColumnTests in test_api that were already covered by test_expr, and moved the rest into that module.

**Comment by patrick-schultz** [comment]
Nice, I think this is a giant improvement. There are still spots where the pairing between source file and test file isn't exact, but hopefully having it most of the way there will encourage better modularization of both the source and tests over time (broken windows and all that).

**Comment by tpoterba** [comment]
yeah, test_expr in particular needs to be broken up

**Comment by tpoterba** [comment]
ah, also -- one more semantic change. I made stopTestHailContext a no-op, so we don't have to reinitialize Spark for every test module. This is OK, right? We have very few modifications to global state -- reference genomes are the only one I can think of.

**Comment by patrick-schultz** [comment]
And conversely, the fact that test_table is split up suggests to me that table.py should probably be split to match. But this pr should probably stick to test reorganization.

**Comment by tpoterba** [comment]
Will follow up with these changes. Prefer to get this in before conflicts pile up.

**Pull request merged**